crates/maybenot-ffi/src/error.rs

```diff
@@ -0,0 +1,47 @@
+use std::str::Utf8Error;
+
+/// An FFI friendly result error code type.
+#[repr(u32)]
+#[derive(Debug, Clone, Copy)]
+pub enum MaybenotResult {
+    /// Operation completed successfully
+    Ok = 0,
+    MachineStringNotUtf8 = 1,
+    InvalidMachineString = 2,
+    StartFramework = 3,
+    NullPointer = 4,
+}
+
+#[derive(Clone, Debug, thiserror::Error)]
+pub enum Error {
```

Comment:
Do we really need to define this error type at all? It is not part of the public API and its "Rust error information" is never printed/returned from what I can see? We just use it as a shim between Rust and the `MaybenotResult` enum, can't we rely solely on the latter? From what I can see the only place we use this error is in the `start` method.

Getting rid of this would also get rid of the `thiserror` dependency here, which would be great if it's not really needed.

Reply:
true